src/main.rs

```diff
@@ -508,6 +510,23 @@ fn run(args: Cli) -> Result<i32> {
 
         let mut request = request_builder.headers(headers).build()?;
 
+        if args.compress >= 1 && request.headers().get(CONTENT_ENCODING).is_none() {
```

Comment:
- If the user passed `--compress` but also a `Content-Encoding` header we should print a warning to let them know that their request isn't compressed and why, just like with `--continue` and `Range`.
  ```rust
  log::warn!("--compress can't be used with a 'Content-Encoding:' header. --compress will be disabled.");
  ```
  Maybe we don't even need to mention it in the docs if we do this, not sure.

- HTTPie actually handles this differently, it forces the compressed Content-Encoding. But I think I like this behavior better. We should still put a comment about the difference.

- I think we can use [`contains_key`](https://docs.rs/reqwest/latest/reqwest/header/struct.HeaderMap.html#method.contains_key) here